test/Clang/clang_shared_library.py

```diff
@@ -41,15 +43,18 @@
     filename = 'foo.os'
     libraryname = 'libfoo.dylib'
 elif platform == 'win32':
-    filename = 'foo.obj'
+    filename = 'foo.os'
     libraryname = 'foo.dll'
+    # add the environment, otherwise the environment will consist of only vcvarsall.bat variables
+    # and not clang
+    env_str = "import os\nenv = Environment(tools=['clang', 'link'], ENV = os.environ)"
```

Comment:
Why do this only for win32?

Reply:
The Environment on windows was created without the original environment that the runtest.py was called from. Not sure if that is an appveyor specific issue or something weird with the msvs tool running vcvarsall.bat, but the mingw tools for linking were missing from the path causing a build failure on the link step during the test.

Adding the os.environ here retained the original environment and path that runtest.py was run from.